gallery/visualtests/nxCombobox.js

```diff
@@ -0,0 +1,210 @@
+        await checkScreenshot(component, 150, 376);
+      });
+
+      it('passes a11Y checks', a11yTest());
+    });
+
+    describe('when invalid', function() {
+      beforeEach(async function() {
+        const inputSelector = `${nonEmptinessExampleSelector} .nx-combobox__input input`,
+            dropdownButtonSelector = `${nonEmptinessExampleSelector} .nx-dropdown-button`,
+            [input] = await waitAndGetElements(inputSelector);
+
+        await input.focus();
+        await getPage().keyboard.type('a');
+        await getPage().waitForSelector(dropdownButtonSelector);
+        await getPage().keyboard.press('Backspace');
+      });
+
+      it('has invalid validation styles when invalid', async function() {
+        const [component] = await waitAndGetElements(nonEmptinessExampleSelector);
+        await checkScreenshot(component, 150, 88);
+      });
+
+      it('passes a11Y checks', a11yTest());
+    });
+  });
+
+  describe('when displaying an error', function() {
+    beforeEach(async function() {
+      const inputSelector = `${errorExampleSelector} .nx-combobox__input input`,
+          errorSelector = `${errorExampleSelector} .nx-alert--load-error`,
+          [input] = await waitAndGetElements(inputSelector);
+
+      await input.focus();
+      await getPage().keyboard.type('1');
+      await getPage().waitForSelector(errorSelector);
+    });
+
+    it('looks right', async function() {
+      const [component] = await waitAndGetElements(errorExampleSelector);
+      await checkScreenshot(component, 300, 195);
+    });
+
+    it('passes a11Y checks', a11yTest());
+  });
+
+  describe('when disabled', function() {
+    it('looks right', simpleTest(disabledExampleSelector));
+  });
+
+  describe('close the dropdown when not focused', function() {
+    it('looks right', async function() {
+      const inputSelector = `${basicExampleSelector} .nx-combobox__input input`,
+          dropdownButtonSelector = `${basicExampleSelector} .nx-dropdown-button`,
+          [component, input] = await waitAndGetElements(basicExampleSelector, inputSelector);
+
+      await input.focus();
+      await getPage().keyboard.type('a');
+      await getPage().waitForSelector(dropdownButtonSelector);
+      await blurElement(input);
+      await checkScreenshot(component, 800, 55);
+    });
+  });
+
+  describe('dropdown button has a blue background when selected', function() {
+    it('looks right', async function() {
+      const inputSelector = `${basicExampleSelector} .nx-combobox__input input`,
+          dropdownButtonSelector = `${basicExampleSelector} .nx-dropdown-button`,
+          [component, input] = await waitAndGetElements(basicExampleSelector, inputSelector);
+
+      await input.focus();
+      await getPage().keyboard.type('a');
+      await getPage().waitForSelector(dropdownButtonSelector);
+      await getPage().keyboard.press('ArrowDown');
+      await checkScreenshot(component, 800, 376);
+    });
+  });
+
```

Comment:
Missing visual tests:
* hover test for the input
* interaction states for the dropdown buttons:
** hover
** focus
** active aka being clicked

Reply:
Updated here 19c5b7eebbe4f4c81fdfe80b34dc05ed35b1cba2.